### ISSUE TITLE
Unsupported inline asm

### ISSUE THREAD

**Issue by sudansh** [cine-io/android-ffmpeg-with-rtmp#25]
MacOS Sierra 10.12

My build.log is below

`Android NDK: WARNING:/Users/sudansh/Desktop/android-ffmpeg-with-rtmp/src/openssl-android/crypto/Android.mk:crypto_static: LOCAL_LDLIBS is always ignored for static libraries    
Android NDK: WARNING:/Users/sudansh/Desktop/android-ffmpeg-with-rtmp/src/openssl-android/crypto/Android.mk:crypto_static: LOCAL_LDLIBS is always ignored for static libraries    
Android NDK: WARNING:/Users/sudansh/Desktop/android-ffmpeg-with-rtmp/src/openssl-android/crypto/Android.mk:crypto_static: LOCAL_LDLIBS is always ignored for static libraries    
Android NDK: WARNING:/Users/sudansh/Desktop/android-ffmpeg-with-rtmp/src/openssl-android/crypto/Android.mk:crypto_static: LOCAL_LDLIBS is always ignored for static libraries    
Android NDK: WARNING:/Users/sudansh/Desktop/android-ffmpeg-with-rtmp/src/openssl-android/crypto/Android.mk:crypto_static: LOCAL_LDLIBS is always ignored for static libraries    
Android NDK: WARNING:/Users/sudansh/Desktop/android-ffmpeg-with-rtmp/src/openssl-android/crypto/Android.mk:crypto_static: LOCAL_LDLIBS is always ignored for static libraries    
Android NDK: WARNING:/Users/sudansh/Desktop/android-ffmpeg-with-rtmp/src/openssl-android/crypto/Android.mk:crypto_static: LOCAL_LDLIBS is always ignored for static libraries    
[arm64-v8a] Install        : libcrypto.so => libs/arm64-v8a/libcrypto.so
[arm64-v8a] Install        : openssl => libs/arm64-v8a/openssl
[arm64-v8a] Install        : libssl.so => libs/arm64-v8a/libssl.so
[arm64-v8a] Install        : ssltest => libs/arm64-v8a/ssltest
[x86_64] Compile        : crypto <= md4_dgst.c
/Users/sudansh/Desktop/android-ffmpeg-with-rtmp/src/openssl-android/crypto/md4/md4_dgst.c:110:5: error: unsupported inline asm: input with type 'unsigned long' matching output with type 'unsigned int'
        R0(A,B,C,D,X( 0), 3,0); HOST_c2l(data,l); X( 2)=l;
        ~~~^~~~~~~~~~~~~~~~~~~
/Users/sudansh/Desktop/android-ffmpeg-with-rtmp/src/openssl-android/crypto/md4/md4_locl.h:104:11: note: expanded from macro 'R0'
        a=ROTATE(a,s); };
          ~~~~~~~^~~~
/Users/sudansh/Desktop/android-ffmpeg-with-rtmp/src/openssl-android/crypto/md32_common.h:168:19: note: expanded from macro 'ROTATE'
                                : "I"(n), "0"(a)        \
                                              ^
/Users/sudansh/Desktop/android-ffmpeg-with-rtmp/src/openssl-android/crypto/md4/md4_dgst.c:111:5: error: unsupported inline asm: input with type 'unsigned long' matching output with type 'unsigned int'
        R0(D,A,B,C,X( 1), 7,0); HOST_c2l(data,l); X( 3)=l;
        ~~~^~~~~~~~~~~~~~~~~~~
/Users/sudansh/Desktop/android-ffmpeg-with-rtmp/src/openssl-android/crypto/md4/md4_locl.h:104:11: note: expanded from macro 'R0'
        a=ROTATE(a,s); };
          ~~~~~~~^~~~
/Users/sudansh/Desktop/android-ffmpeg-with-rtmp/src/openssl-android/crypto/md32_common.h:168:19: note: expanded from macro 'ROTATE'
                                : "I"(n), "0"(a)        \
                                              ^
/Users/sudansh/Desktop/android-ffmpeg-with-rtmp/src/openssl-android/crypto/md4/md4_dgst.c:112:5: error: unsupported inline asm: input with type 'unsigned long' matching output with type 'unsigned int'
        R0(C,D,A,B,X( 2),11,0); HOST_c2l(data,l); X( 4)=l;
        ~~~^~~~~~~~~~~~~~~~~~~
/Users/sudansh/Desktop/android-ffmpeg-with-rtmp/src/openssl-android/crypto/md4/md4_locl.h:104:11: note: expanded from macro 'R0'
        a=ROTATE(a,s); };
          ~~~~~~~^~~~
/Users/sudansh/Desktop/android-ffmpeg-with-rtmp/src/openssl-android/crypto/md32_common.h:168:19: note: expanded from macro 'ROTATE'
                                : "I"(n), "0"(a)        \
                                              ^
/Users/sudansh/Desktop/android-ffmpeg-with-rtmp/src/openssl-android/crypto/md4/md4_dgst.c:113:5: error: unsupported inline asm: input with type 'unsigned long' matching output with type 'unsigned int'
        R0(B,C,D,A,X( 3),19,0); HOST_c2l(data,l); X( 5)=l;
        ~~~^~~~~~~~~~~~~~~~~~~
/Users/sudansh/Desktop/android-ffmpeg-with-rtmp/src/openssl-android/crypto/md4/md4_locl.h:104:11: note: expanded from macro 'R0'
        a=ROTATE(a,s); };
          ~~~~~~~^~~~
/Users/sudansh/Desktop/android-ffmpeg-with-rtmp/src/openssl-android/crypto/md32_common.h:168:19: note: expanded from macro 'ROTATE'
                                : "I"(n), "0"(a)        \
                                              ^
/Users/sudansh/Desktop/android-ffmpeg-with-rtmp/src/openssl-android/crypto/md4/md4_dgst.c:114:5: error: unsupported inline asm: input with type 'unsigned long' matching output with type 'unsigned int'
        R0(A,B,C,D,X( 4), 3,0); HOST_c2l(data,l); X( 6)=l;
        ~~~^~~~~~~~~~~~~~~~~~~
/Users/sudansh/Desktop/android-ffmpeg-with-rtmp/src/openssl-android/crypto/md4/md4_locl.h:104:11: note: expanded from macro 'R0'
        a=ROTATE(a,s); };
          ~~~~~~~^~~~
/Users/sudansh/Desktop/android-ffmpeg-with-rtmp/src/openssl-android/crypto/md32_common.h:168:19: note: expanded from macro 'ROTATE'
                                : "I"(n), "0"(a)        \
                                              ^
/Users/sudansh/Desktop/android-ffmpeg-with-rtmp/src/openssl-android/crypto/md4/md4_dgst.c:115:5: error: unsupported inline asm: input with type 'unsigned long' matching output with type 'unsigned int'
        R0(D,A,B,C,X( 5), 7,0); HOST_c2l(data,l); X( 7)=l;
        ~~~^~~~~~~~~~~~~~~~~~~
/Users/sudansh/Desktop/android-ffmpeg-with-rtmp/src/openssl-android/crypto/md4/md4_locl.h:104:11: note: expanded from macro 'R0'
        a=ROTATE(a,s); };
          ~~~~~~~^~~~
/Users/sudansh/Desktop/android-ffmpeg-with-rtmp/src/openssl-android/crypto/md32_common.h:168:19: note: expanded from macro 'ROTATE'
                                : "I"(n), "0"(a)        \
                                              ^
/Users/sudansh/Desktop/android-ffmpeg-with-rtmp/src/openssl-android/crypto/md4/md4_dgst.c:116:5: error: unsupported inline asm: input with type 'unsigned long' matching output with type 'unsigned int'
        R0(C,D,A,B,X( 6),11,0); HOST_c2l(data,l); X( 8)=l;
        ~~~^~~~~~~~~~~~~~~~~~~
/Users/sudansh/Desktop/android-ffmpeg-with-rtmp/src/openssl-android/crypto/md4/md4_locl.h:104:11: note: expanded from macro 'R0'
        a=ROTATE(a,s); };
          ~~~~~~~^~~~
/Users/sudansh/Desktop/android-ffmpeg-with-rtmp/src/openssl-android/crypto/md32_common.h:168:19: note: expanded from macro 'ROTATE'
                                : "I"(n), "0"(a)        \
                                              ^
/Users/sudansh/Desktop/android-ffmpeg-with-rtmp/src/openssl-android/crypto/md4/md4_dgst.c:117:5: error: unsupported inline asm: input with type 'unsigned long' matching output with type 'unsigned int'
        R0(B,C,D,A,X( 7),19,0); HOST_c2l(data,l); X( 9)=l;
        ~~~^~~~~~~~~~~~~~~~~~~
/Users/sudansh/Desktop/android-ffmpeg-with-rtmp/src/openssl-android/crypto/md4/md4_locl.h:104:11: note: expanded from macro 'R0'
        a=ROTATE(a,s); };
          ~~~~~~~^~~~
/Users/sudansh/Desktop/android-ffmpeg-with-rtmp/src/openssl-android/crypto/md32_common.h:168:19: note: expanded from macro 'ROTATE'
                                : "I"(n), "0"(a)        \
                                              ^
/Users/sudansh/Desktop/android-ffmpeg-with-rtmp/src/openssl-android/crypto/md4/md4_dgst.c:118:5: error: unsupported inline asm: input with type 'unsigned long' matching output with type 'unsigned int'
        R0(A,B,C,D,X( 8), 3,0); HOST_c2l(data,l); X(10)=l;
        ~~~^~~~~~~~~~~~~~~~~~~
/Users/sudansh/Desktop/android-ffmpeg-with-rtmp/src/openssl-android/crypto/md4/md4_locl.h:104:11: note: expanded from macro 'R0'
        a=ROTATE(a,s); };
          ~~~~~~~^~~~
/Users/sudansh/Desktop/android-ffmpeg-with-rtmp/src/openssl-android/crypto/md32_common.h:168:19: note: expanded from macro 'ROTATE'
                                : "I"(n), "0"(a)        \
                                              ^
/Users/sudansh/Desktop/android-ffmpeg-with-rtmp/src/openssl-android/crypto/md4/md4_dgst.c:119:5: error: unsupported inline asm: input with type 'unsigned long' matching output with type 'unsigned int'
        R0(D,A,B,C,X( 9), 7,0); HOST_c2l(data,l); X(11)=l;
        ~~~^~~~~~~~~~~~~~~~~~~
/Users/sudansh/Desktop/android-ffmpeg-with-rtmp/src/openssl-android/crypto/md4/md4_locl.h:104:11: note: expanded from macro 'R0'
        a=ROTATE(a,s); };
          ~~~~~~~^~~~
/Users/sudansh/Desktop/android-ffmpeg-with-rtmp/src/openssl-android/crypto/md32_common.h:168:19: note: expanded from macro 'ROTATE'
                                : "I"(n), "0"(a)        \
                                              ^
/Users/sudansh/Desktop/android-ffmpeg-with-rtmp/src/openssl-android/crypto/md4/md4_dgst.c:120:5: error: unsupported inline asm: input with type 'unsigned long' matching output with type 'unsigned int'
        R0(C,D,A,B,X(10),11,0); HOST_c2l(data,l); X(12)=l;
        ~~~^~~~~~~~~~~~~~~~~~~
/Users/sudansh/Desktop/android-ffmpeg-with-rtmp/src/openssl-android/crypto/md4/md4_locl.h:104:11: note: expanded from macro 'R0'
        a=ROTATE(a,s); };
          ~~~~~~~^~~~
/Users/sudansh/Desktop/android-ffmpeg-with-rtmp/src/openssl-android/crypto/md32_common.h:168:19: note: expanded from macro 'ROTATE'
                                : "I"(n), "0"(a)        \
                                              ^
/Users/sudansh/Desktop/android-ffmpeg-with-rtmp/src/openssl-android/crypto/md4/md4_dgst.c:121:5: error: unsupported inline asm: input with type 'unsigned long' matching output with type 'unsigned int'
        R0(B,C,D,A,X(11),19,0); HOST_c2l(data,l); X(13)=l;
        ~~~^~~~~~~~~~~~~~~~~~~
/Users/sudansh/Desktop/android-ffmpeg-with-rtmp/src/openssl-android/crypto/md4/md4_locl.h:104:11: note: expanded from macro 'R0'
        a=ROTATE(a,s); };
          ~~~~~~~^~~~
/Users/sudansh/Desktop/android-ffmpeg-with-rtmp/src/openssl-android/crypto/md32_common.h:168:19: note: expanded from macro 'ROTATE'
                                : "I"(n), "0"(a)        \
                                              ^
/Users/sudansh/Desktop/android-ffmpeg-with-rtmp/src/openssl-android/crypto/md4/md4_dgst.c:122:5: error: unsupported inline asm: input with type 'unsigned long' matching output with type 'unsigned int'
        R0(A,B,C,D,X(12), 3,0); HOST_c2l(data,l); X(14)=l;
        ~~~^~~~~~~~~~~~~~~~~~~
/Users/sudansh/Desktop/android-ffmpeg-with-rtmp/src/openssl-android/crypto/md4/md4_locl.h:104:11: note: expanded from macro 'R0'
        a=ROTATE(a,s); };
          ~~~~~~~^~~~
/Users/sudansh/Desktop/android-ffmpeg-with-rtmp/src/openssl-android/crypto/md32_common.h:168:19: note: expanded from macro 'ROTATE'
                                : "I"(n), "0"(a)        \
                                              ^
/Users/sudansh/Desktop/android-ffmpeg-with-rtmp/src/openssl-android/crypto/md4/md4_dgst.c:123:5: error: unsupported inline asm: input with type 'unsigned long' matching output with type 'unsigned int'
        R0(D,A,B,C,X(13), 7,0); HOST_c2l(data,l); X(15)=l;
        ~~~^~~~~~~~~~~~~~~~~~~
/Users/sudansh/Desktop/android-ffmpeg-with-rtmp/src/openssl-android/crypto/md4/md4_locl.h:104:11: note: expanded from macro 'R0'
        a=ROTATE(a,s); };
          ~~~~~~~^~~~
/Users/sudansh/Desktop/android-ffmpeg-with-rtmp/src/openssl-android/crypto/md32_common.h:168:19: note: expanded from macro 'ROTATE'
                                : "I"(n), "0"(a)        \
                                              ^
/Users/sudansh/Desktop/android-ffmpeg-with-rtmp/src/openssl-android/crypto/md4/md4_dgst.c:124:5: error: unsupported inline asm: input with type 'unsigned long' matching output with type 'unsigned int'
        R0(C,D,A,B,X(14),11,0);
        ~~~^~~~~~~~~~~~~~~~~~~
/Users/sudansh/Desktop/android-ffmpeg-with-rtmp/src/openssl-android/crypto/md4/md4_locl.h:104:11: note: expanded from macro 'R0'
        a=ROTATE(a,s); };
          ~~~~~~~^~~~
/Users/sudansh/Desktop/android-ffmpeg-with-rtmp/src/openssl-android/crypto/md32_common.h:168:19: note: expanded from macro 'ROTATE'
                                : "I"(n), "0"(a)        \
                                              ^
/Users/sudansh/Desktop/android-ffmpeg-with-rtmp/src/openssl-android/crypto/md4/md4_dgst.c:125:5: error: unsupported inline asm: input with type 'unsigned long' matching output with type 'unsigned int'
        R0(B,C,D,A,X(15),19,0);
        ~~~^~~~~~~~~~~~~~~~~~~
/Users/sudansh/Desktop/android-ffmpeg-with-rtmp/src/openssl-android/crypto/md4/md4_locl.h:104:11: note: expanded from macro 'R0'
        a=ROTATE(a,s); };
          ~~~~~~~^~~~
/Users/sudansh/Desktop/android-ffmpeg-with-rtmp/src/openssl-android/crypto/md32_common.h:168:19: note: expanded from macro 'ROTATE'
                                : "I"(n), "0"(a)        \
                                              ^
/Users/sudansh/Desktop/android-ffmpeg-with-rtmp/src/openssl-android/crypto/md4/md4_dgst.c:127:5: error: unsupported inline asm: input with type 'unsigned long' matching output with type 'unsigned int'
        R1(A,B,C,D,X( 0), 3,0x5A827999L);
        ~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/Users/sudansh/Desktop/android-ffmpeg-with-rtmp/src/openssl-android/crypto/md4/md4_locl.h:108:11: note: expanded from macro 'R1'
        a=ROTATE(a,s); };\
          ~~~~~~~^~~~
/Users/sudansh/Desktop/android-ffmpeg-with-rtmp/src/openssl-android/crypto/md32_common.h:168:19: note: expanded from macro 'ROTATE'
                                : "I"(n), "0"(a)        \
                                              ^
/Users/sudansh/Desktop/android-ffmpeg-with-rtmp/src/openssl-android/crypto/md4/md4_dgst.c:128:5: error: unsupported inline asm: input with type 'unsigned long' matching output with type 'unsigned int'
        R1(D,A,B,C,X( 4), 5,0x5A827999L);
        ~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/Users/sudansh/Desktop/android-ffmpeg-with-rtmp/src/openssl-android/crypto/md4/md4_locl.h:108:11: note: expanded from macro 'R1'
        a=ROTATE(a,s); };\
          ~~~~~~~^~~~
/Users/sudansh/Desktop/android-ffmpeg-with-rtmp/src/openssl-android/crypto/md32_common.h:168:19: note: expanded from macro 'ROTATE'
                                : "I"(n), "0"(a)        \
                                              ^
/Users/sudansh/Desktop/android-ffmpeg-with-rtmp/src/openssl-android/crypto/md4/md4_dgst.c:129:5: error: unsupported inline asm: input with type 'unsigned long' matching output with type 'unsigned int'
        R1(C,D,A,B,X( 8), 9,0x5A827999L);
        ~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/Users/sudansh/Desktop/android-ffmpeg-with-rtmp/src/openssl-android/crypto/md4/md4_locl.h:108:11: note: expanded from macro 'R1'
        a=ROTATE(a,s); };\
          ~~~~~~~^~~~
/Users/sudansh/Desktop/android-ffmpeg-with-rtmp/src/openssl-android/crypto/md32_common.h:168:19: note: expanded from macro 'ROTATE'
                                : "I"(n), "0"(a)        \
                                              ^
fatal error: too many errors emitted, stopping now [-ferror-limit=]
20 errors generated.
make: *** [obj/local/x86_64/objs/crypto/md4/md4_dgst.o] Error 1
`

**Comment by xdeop** [comment]
Hi,
Same for me with . Any solution?
Thanks.

**Comment by lomoya90** [comment]
Me too.Any solution?

**Comment by rashikaranpuria** [comment]
Having the same issue. Any progress you guys?

**Comment by rpocaznoi** [comment]
I encountered this exact problem with the ROTATE macro recently when trying to build openssl 1.0.2l for iPhone simulator x86_64 architecture.
Apparently some compilers have trouble handling the inline asm defined by the ROTATE macro (I see it defined in des_locl.h and md32_common.h in my openssl sources).

As a workaround I've added `-no-inline-asm` as additional argument in the openssl's ./Configure line and this forced the ROTATE macro to the non-asm fallback version and it built fine.
